### ISSUE TITLE
Incorrect GIT commit info for lines (author/committed on/revision) for branch analysis with this plugin in SonarQube community version 

### ISSUE THREAD

**Issue by zhaoan2** [mc1arke/sonarqube-community-branch-plugin#832]
Hi all,

### **Describe the bug**
Found the GIT author/committed on/revision information is incorrect by using this plugin for branch analysis in SonarQube community version.

I have two SonarQube servers (same version) for difference usages：
**SonarQube1** does not use this plugin, it uses default scan mode with new code setting "Specific analysis". (it does not distinguish any branch and does not integrate with any CI, set all analysis to default "main" branch in SonarQube).
**SonarQube2** is for branch analysis (set the new code setting of Branch 2 to Branch1 as its "Reference Branch" ) with this plugin.

After comparation, I found the GIT author/committed on/revision information showing in the SonarQube 2 for line level are sometimes incorrect.

**For example:** 
Same class, same line:
SonarQube1 shows correct GIT info for line 26:
```
Author: mxxx
Committed on: Nov 22, 2023
Revision: 218xxx
```

SonarQube2 shows correct GIT info for line 26: (branch = Branch1)
```
Author: kxxx
Committed on: Aug 03, 2023
Revision: 036xxx
```

SonarQube2 shows **incorrect** GIT info for line 26: (branch = Branch2)
```
Author: kxxx
Committed on: Aug 03, 2023
Revision: 036xxx
```

What is it in GIT? => This line was committed by kxxx on Aug 03 in Branch1 but re-committed by mxxx on Nov 22 in Branch2. 

Would you please suggest how can I get the correct line GIT info shown in SonarQube dashboard for new code analysis with "Reference Branch" analysis mode with this plugin? Thank a lot.

BTW, I have never tried in SonarQube Developer/Enterprise Edition.

### **Software Versions**

SonarQube Version: 9.9.1
Plugin Version: 1.14.0

Thanks for your help!!!

**Comment by zhaoanna** [comment]
Hi all,
Good news! Seems I found how to resolve this issue. It is about the "MAIN BRANCH" setting in my SonarQube2.
In my case, for "Branch Analysis" in SonarQube2, I have set the branch1 as "MAIN BRANCH" (name of branch1 is not main or master). Now I resolved this issue by "Not set my branch to 'MAIN BRANCH'" and the GIT commit info of new code lines were back to normal.

But does anyone have any idea that why the "MAIN BRANCH" setting impacts to the GIT commit info? I did not find any related information about this in SonarQube official documentation.

**Comment by zhaoan2** [comment]
Hi all,
**Good news!** Seems I found how to resolve this issue. It is about the "MAIN BRANCH" setting in my SonarQube2.
In my case, for "Branch Analysis" in SonarQube2, I have set the branch1 as "MAIN BRANCH" (name of branch1 is not main or master). Now I resolved this issue by "Not set my branch to 'MAIN BRANCH'" and the GIT commit info of new code lines were back to normal.

But does anyone have any idea that why the "MAIN BRANCH" setting impacts to the GIT commit info? I did not find any related information about this in SonarQube official documentation.
